src/libclipper/include/clipper/logging.hpp

```diff
@@ -0,0 +1,172 @@
+#ifndef CLIPPER_LOGGING_HPP
+#define CLIPPER_LOGGING_HPP
+
+#include <string>
+#include <sstream>
+
+#include <spdlog/spdlog.h>
+
+namespace clipper {
+
+static constexpr uint LOGGING_FORMAT_LENGTH = 18;
+static constexpr uint MAX_TAG_LENGTH = 10;
+// Defines the logging format as [HH:MM:SS.mmm][<LOG_LEVEL_1_CHAR>] <Message>
```

Comment:
Done! This required a few changes in order to preserve alignment. I've created a `LogLevel` enum and several constants defining the lengths of the different log level tags. These values are used to pad log messages of different levels correctly.